benchmark-service/benchmark/benchmark.py

```diff
@@ -2,19 +2,78 @@
 import aiohttp_jinja2
 from aiohttp import web
 import logging
-from gear import setup_aiohttp_session, web_authenticated_developers_only
+from gear import setup_aiohttp_session, web_authenticated_developers_only, check_csrf_token
 from hailtop.config import get_deploy_config
 from hailtop.tls import get_in_cluster_server_ssl_context
 from hailtop.hail_logging import AccessLogger, configure_logging
-from web_common import setup_aiohttp_jinja2, setup_common_static_routes
+from web_common import setup_aiohttp_jinja2, setup_common_static_routes, render_template
 
+import json
+import re
+from google.cloud import storage
+import plotly
+import plotly.express as px
 
 configure_logging()
 router = web.RouteTableDef()
 logging.basicConfig(level=logging.DEBUG)
 deploy_config = get_deploy_config()
 log = logging.getLogger('benchmark')
 
+FILE_PATH_REGEX = re.compile(r'((?P<user>[^/]+)/)((?P<version>[^-]+)-)((?P<sha>[^-]+))(-(?P<tag>[^\.]+))?\.json')
+filepath = 'tpoterba/0.2.45-ac6815ee857c-master.json'
+
+
+def parse_file_path(name):
+    match = FILE_PATH_REGEX.fullmatch(name)
+    return match.groupdict()
+
+
+def get_benchmarks(file_path):
+    # create storage client
+    storage_client = storage.Client()
+    # get bucket with name
+    bucket = storage_client.get_bucket('hail-benchmarks')
+    try:
+        # get bucket data as blob
+        blob = bucket.blob(file_path)
+        # convert to string
+        json_data = blob.download_as_string()
+        pre_data = json.loads(json_data)
+    except Exception:
+        message = 'could not find file'
+        log.info('could not get blob: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message)
+
+    # x = re.findall('.*/+(.*)-(.*)-(.*)?\.json', file_path)
+    # sha = x[0][1]
+
+    file_info = parse_file_path(file_path)
+    sha = file_info['sha']
+
+    data = list()
+    prod_of_means = 1
+    for d in pre_data['benchmarks']:
+        stats = dict()
+        stats['name'] = d['name']
+        stats['failed'] = d['failed']
+        if not d['failed']:
+            prod_of_means *= d['mean']
+            stats['f-stat'] = round(d['f-stat'], 6)
+            stats['mean'] = round(d['mean'], 6)
+            stats['median'] = round(d['median'], 6)
+            stats['p-value'] = round(d['p-value'], 6)
+            stats['stdev'] = round(d['stdev'], 6)
+            stats['times'] = d['times']
+        data.append(stats)
+    geometric_mean = prod_of_means ** (1.0 / len(pre_data['benchmarks']))
```

Comment:
Let's move this equation to the utils.py file and make it a function that you can call.